src/com/owncloud/android/datamodel/OCFile.java

```diff
@@ -747,4 +749,36 @@ public void copyLocalPropertiesFrom(OCFile sourceFile) {
         setEtagInConflict(sourceFile.getEtagInConflict());
     }
 
+    /**
+     * Manipulate file remote id to get private link
+     *
+     * @param context
+     * @param account
+     * @return private link for the current file
+     */
+    public String getPrivateLink(Context context, Account account) {
+
+        // Parse remoteId
+        String parsedRemoteId = mRemoteId.substring(0, Math.min(mRemoteId.length(), 8));
+
+        String fileId = Integer.valueOf(parsedRemoteId).toString();
+
+        String privateLink = null;
+
+        try {
+
+            String baseUrl = AccountUtils.getBaseUrlForAccount(context, account);
+
+            if (baseUrl == null)
+                throw new AccountUtils.AccountNotFoundException(account, "Account not found", null);
+
+            privateLink = baseUrl + AccountUtils.INDEX_PATH + FILES_PATH + "/" + fileId;
```

Comment:
I've separated it into two different paths just in case we need to use /index.php in the future.